Assets/Mirror/Runtime/Batching/Batcher.cs

```diff
@@ -45,53 +51,75 @@ public Batcher(int threshold)
         // add a message for batching
         // we allow any sized messages.
         // caller needs to make sure they are within max packet size.
-        public void AddMessage(ArraySegment<byte> message)
+        public void AddMessage(ArraySegment<byte> message, double timeStamp)
         {
-            // put into a (pooled) writer
+            // when appending to a batch in progress, check final size.
+            // if it expands beyond threshold, then we should finalize it first.
+            // => less than or exactly threshold is fine.
+            //    GetBatch() will finalize it.
+            // => see unit tests.
+            if (batch != null &&
+                batch.Position + message.Count > threshold)
+            {
+                batches.Enqueue(batch);
+                batch = null;
+            }
+
+            // initialize a new batch if necessary
+            if (batch == null)
```

Comment:
Do we need special handling here if the timestamp is different to make a new batch?
(that might break on 2019 though where we still use the running stopwatch timer so it'd always be different)